client/my-sites/checkout/controller.jsx

```diff
@@ -140,7 +140,11 @@ export function checkout( context, next ) {
 	const couponCode = context.query.coupon || context.query.code || getRememberedCoupon();
 
 	const isLoggedOutCart =
-		isJetpackCheckout || ( isLoggedOut && context.pathname.includes( '/checkout/no-site' ) );
+		isJetpackCheckout ||
+		( isLoggedOut && context.pathname.includes( '/checkout/no-site' ) ) ||
+		( isLoggedOut &&
+			context.pathname.includes( '/gift/' ) &&
+			context.pathname.startsWith( '/checkout/' ) );
```

Comment:
Do we need to check that the path starts with /checkout, or will the /gift path always be a checkout path?

Reply:
Sounds good to remove, we are already at a /checkout page :)